chain/store/store.go

```diff
@@ -863,10 +863,16 @@ func (cs *ChainStore) Contains(ctx context.Context, ts *types.TipSet) (bool, err
 func (cs *ChainStore) GetBlock(ctx context.Context, c cid.Cid) (*types.BlockHeader, error) {
 	var blk *types.BlockHeader
 	err := cs.chainLocalBlockstore.View(ctx, c, func(b []byte) (err error) {
+		if len(b) == 0 {
+			return nil // Block not found
+		}
 		blk, err = types.DecodeBlock(b)
 		return err
 	})
-	return blk, err
+	if err != nil {
+		return nil, err
+	}
+	return blk, nil
```

Comment:
```suggestion
	if blk == nil {
		return nil, nil
	}
	return blk, nil
```

small nit, let's not return the typed `nil`, it can have surprising effects in the call chain where comparisons might come in to play